engine/transport/factory.go

```diff
@@ -0,0 +1,21 @@
+package transport
+
+import (
+	"fmt"
+	"time"
+)
+
+var creators = make(map[string]Creator)
+
+func Register(name string, creator Creator) {
+	creators[name] = creator
```

Comment:
Maybe should add lock,i did't trust user and golang map

Reply:
All Register() should be called before creating any transports. Basically, it will be in `init()`. I don't think it's necessary to lock the map here.